gralloc_gbm.cpp

```diff
@@ -529,3 +543,54 @@ int gralloc_gbm_bo_lock_ycbcr(buffer_handle_t handle,
 
 	return 0;
 }
+
+/*
+ * Check if target device has KMS.
+ */
+bool is_kms_dev(int fd)
+{
+	auto res = drmModeGetResources(fd);
+	if (!res)
+		return false;
+
+	bool is_kms = res->count_crtcs > 0 && res->count_connectors > 0 &&
+			res->count_encoders > 0;
+
+	drmModeFreeResources(res);
+
+	return is_kms;
+}
+
+/*
+ * Search for a KMS device. Return opened file descriptor on success.
+ */
+int open_first_kms_dev(const char *pattern)
+{
+	glob_t glob_buf;
+	memset(&glob_buf, 0, sizeof(glob_buf));
+	int fd;
```

Comment:
Will `glob()` return `0` (instead of an error-code) if it found zero matching paths? If so, this should be initialized as to not return garbage when the loop never runs.